src/common/models/Projection.ts

```diff
@@ -252,35 +232,40 @@ export class Projection {
     );
 
     const disableRt = false;
-    this.rtRange = disableRt ? [null] : this.calcRtRange(timeseries);
-    this.testPositiveRate = this.calcTestPositiveRate();
-
-    this.icuHeadroomInfo = calcICUHeadroom(
-      this.stateName,
-      this.fips,
-      this.dates,
-      actualTimeseries,
-      timeseries,
-      actuals,
-      lastUpdated,
+    this.rtRange = disableRt ? [null] : this.calcRtRange(metricsTimeseries);
+    this.testPositiveRate = metricsTimeseries.map(row =>
+      row ? row.testPositivityRatio : null,
     );
+
+    if (metrics) {
+      this.icuHeadroomInfo = calcICUHeadroom(
+        this.fips,
+        actualTimeseries,
+        metricsTimeseries,
+        metrics,
+        actuals,
+      );
+    } else {
+      this.icuHeadroomInfo = null;
+    }
     this.icuUtilization =
       this.icuHeadroomInfo?.metricSeries || this.dates.map(date => null);
 
-    this.contractTracers = this.calcContactTracers(this.actualTimeseries);
+    this.contractTracers = metricsTimeseries.map(row =>
+      row ? row.contactTracerCapacityRatio : null,
+    );
 
-    this.caseDensityByCases = this.calcCaseDensityByCases();
+    this.caseDensityByCases = metricsTimeseries.map(row =>
+      row ? row.caseDensity : null,
+    );
     this.caseDensityByDeaths = this.calcCaseDensityByDeaths();
     this.caseDensityRange = this.calcCaseDensityRange();
 
-    this.currentCaseDensityByCases = lastValue(this.caseDensityByCases);
+    this.currentCaseDensityByCases = metrics ? metrics!.caseDensity : null;
```

Comment:
`!` shouldn't be necessary?

Also, FWIW we could clean up this code and get rid of caseDensityByCases/Deaths since I don't think we'll bring deaths back.

Reply:
Aha - yes this is right, I think i was using `metrics?.caseDensity` before not realizing that the first part of the check unwraps the `metrics` variable